enterprise/frontend/src/metabase-enterprise/whitelabel/components/IllustrationWidget/IllustrationWidget.tsx

```diff
@@ -22,62 +24,69 @@ export interface StringSetting {
   default: IllustrationSettingValue;
 }
 
+type IllustrationType = "background" | "icon";
+
 type IllustrationWidgetProps = {
   id?: string;
   setting: StringSetting;
   onChange: (value: IllustrationSettingValue) => Promise<void>;
   onChangeSetting: (key: EnterpriseSettingKey, value: unknown) => Promise<void>;
   settingValues: Partial<EnterpriseSettings>;
-  defaultIllustrationLabel: string;
   customIllustrationSetting:
     | "login-page-illustration-custom"
     | "landing-page-illustration-custom"
     | "no-question-results-illustration-custom"
     | "no-search-results-illustration-custom";
   errorMessageContainerId: string;
-  defaultPreviewType: "lighthouse" | "sailboat";
+  type: IllustrationType;
 };
 
 const MB = 1024 * 1024;
 const IMAGE_SIZE_LIMIT = 2 * MB;
 
+const OPTIONS: Record<IllustrationType, any[]> = {
+  background: [
+    { label: t`Lighthouse`, value: "default" },
+    { label: t`No illustration`, value: "no-illustration" },
+    { label: t`Custom`, value: "custom" },
+  ],
+  icon: [
+    { label: t`Sailboat`, value: "default" },
+    { label: t`No illustration`, value: "no-illustration" },
+    { label: t`Custom`, value: "custom" },
+  ],
+};
+
 export function IllustrationWidget({
   id,
   setting,
   onChange,
   onChangeSetting,
   settingValues,
-  defaultIllustrationLabel,
   customIllustrationSetting,
   errorMessageContainerId,
-  defaultPreviewType,
+  type,
 }: IllustrationWidgetProps) {
-  const value = setting.value ?? setting.default;
+  const [value, setValue] = useState(setting.value ?? setting.default);
```

Comment:
Change to controlled value because when selecting `Custom` when no custom image is uploaded, we shouldn't change any setting, but only show the custom image UI that allows user to upload a custom image.